src/bitcoin-cli.cpp

```diff
@@ -288,7 +288,26 @@ int CommandLineRPC(int argc, char *argv[])
         if (args.size() < 1)
             throw runtime_error("too few parameters (need at least command)");
         std::string strMethod = args[0];
-        UniValue params = RPCConvertValues(strMethod, std::vector<std::string>(args.begin()+1, args.end()));
+        args.erase(args.begin());
+
+        // Check for named arguments
+        bool haveNamed = false;
+        bool havePositional = false;
+        for (const std::string &s: args) {
+            if (s.size() >= 1 && s[0] == '-')
+                haveNamed = true; // Have named arguments
+            else
+                havePositional = true; // Have positional arguments
+        }
+
+        UniValue params;
+        if (haveNamed && havePositional) {
+            throw runtime_error("cannot have both named and positional arguments, did you mean to pass arguments before the command?");
+        } else if(haveNamed) {
+            params = RPCConvertNamedValues(strMethod, args);
```

Comment:
Mixing args here with GetBoolArg/GetArg is a little scary. If some command took a (for example) "-version=foo" argument, I believe it would be swallowed by line 81 instead?